web/src/components/apps/AppVersionHistory.jsx

```diff
@@ -1002,7 +1002,7 @@ class AppVersionHistory extends Component {
       });
 
       const res = await fetch(
-        `${process.env.API_ENDPOINT}/app/${app?.slug}/cluster/${clusterId}/sequence/${version?.sequence}/downstreamoutput`,
+        `${process.env.API_ENDPOINT}/app/${app.slug}/cluster/${clusterId}/sequence/${version.sequence}/downstreamoutput`,
```

Comment:
Slug and version are required here.  If they are missing, UI state is not managed right.